polaris/experimental/_dataset_v2.py

```diff
@@ -264,7 +278,21 @@ def to_json(
             json.dump(serialized, f)
         return dataset_path
 
+    def cache(self, verify_checksum: bool = True) -> str:
```

Comment:
You shouldn't change the function signature when overriding it in a subclass. But since it basically calls `self.to_json` and returns path, I can push it down to the subclasses.